InformationScripting/src/visualization/DefaultVisualizer.cpp

```diff
@@ -80,27 +80,23 @@ void DefaultVisualizer::visualize(const TupleSet& ts)
 	}
 	else
 	{
-		auto allTuples = ts.tuples([](const Tuple& t) { return t.find("ast") != t.end(); });
+		auto astTuples = ts.tuples([](const Tuple& t) { return t.tag() == "ast"; });
 
 		QHash<Model::Node*, QString> colors;
 
-		for (auto t : allTuples)
+		for (auto t : astTuples)
 		{
-			auto allAsts = t.getAll("ast");
-			for (auto astProperty : allAsts)
+			Model::Node* node = t["ast"];
+			QString& color = colors[node];
+			if (color.isEmpty())
 			{
-				Model::Node* node = astProperty.second;
-				QString& color = colors[node];
-				if (color.isEmpty())
+				if (t.find("color") != t.end())
```

Comment:
You're calling find twice here. Don't do that, just remember the iterator and use it later.